homeassistant/components/shelly/config_flow.py

```diff
@@ -310,3 +323,50 @@ async def _async_get_info(self, host: str) -> dict[str, Any]:
         return await aioshelly.common.get_info(
             aiohttp_client.async_get_clientsession(self.hass), host
         )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> OptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    @classmethod
+    @callback
+    def async_supports_options_flow(
+        cls, config_entry: config_entries.ConfigEntry
+    ) -> bool:
+        """Return options flow support for this handler."""
+        return config_entry.data.get("gen") == 2
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle the option flow for shelly."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle options flow."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
```

Comment:
Should we check the version here and tell the user it should update it's Shelly first?

Reply:
will now abort if they choose active or passive and the firmware is too old